src/ralph/models/selector.py

```diff
@@ -64,20 +65,36 @@ class ModelSelector:
         decision_tree (dict): Stores the rule checking order for model selection.
     """
 
-    model_rules: dict[Type[BaseModel], list[Rule]] = {}
-
     def __init__(self, module="ralph.models.edx"):
         """Instantiates ModelSelector."""
 
+        self.model_rules: dict[Type[BaseModel], list[Rule]] = {}
         for object_ in import_module(module).__dict__.values():
             if (
                 isinstance(object_, type)
                 and issubclass(object_, BaseModel)
                 and hasattr(object_, "__selector__")
             ):
-                self.model_rules[object_] = object_.__selector__
+                self.add_to_model_rules(object_, object_.__selector__)
         self.decision_tree = self.get_decision_tree(self.model_rules)
 
+    def add_to_model_rules(self, new_model: Type[BaseModel], new_rules: list[Rule]):
```

Comment:
I would suggest to isolate this utility in a static method and make it more testable by including modules loading:

```suggestion
    @staticmethod
    def build_model_rules(module):
```

Reply:
Good Idea) Thanks) Updated the PR with the build_model_rules method and added a test.